include/seqan3/io/alignment_file/input_format_concept.hpp

```diff
@@ -30,11 +30,82 @@
 namespace seqan3::detail
 {
 
+/*!\brief Internal class used to expose the actual format interface to read alignment records from the file.
+ * \ingroup alignment_file
+ *
+ * \tparam format_type The type of the format to be exposed.
+ *
+ * \details
+ *
+ * Exposes the protected member function `read_alignment_record` from the given `format_type`, such that the file can
+ * call the proper function for the selected format.
+ */
 template <typename format_type>
 struct alignment_file_input_format_exposer : public format_type
 {
 public:
-    using format_type::read_alignment_record;
+    // Can't use `using format_type::read_alignment_record` as it produces a hard failure in the format concept check
```

Comment:
phew, this is ugly.

Can you try to revert this and instead change the concept below to the following:

```diff
SEQAN3_CONCEPT alignment_file_input_format =
-   requires (detail::alignment_file_input_format_exposer<t>                      & v,
-             std::ifstream                                                       & stream,
+   requires (std::ifstream                                                       & stream,
              alignment_file_input_options<dna5>                                  & options,
              std::vector<dna5_vector>                                            & ref_sequences,
              alignment_file_header<>                                             & header,
              dna5_vector                                                         & seq,
              std::vector<phred42>                                                & qual,
              std::string                                                         & id,
              int32_t                                                             & offset,
              dna5_vector                                                         & ref_seq,
              std::optional<int32_t>                                              & ref_id,
              std::optional<int32_t>                                              & ref_offset,
              std::pair<std::vector<gapped<dna4>>, std::vector<gapped<dna4>>>     & align,
              std::vector<cigar>                                                  & cigar,
              uint16_t                                                            & flag,
              uint8_t                                                             & mapq,
              std::tuple<std::optional<int32_t>, std::optional<int32_t>, int32_t> & mate,
              sam_tag_dictionary                                                  & tag_dict,
              double                                                              & e_value,
              double                                                              & bit_score)
{
    t::file_extensions;
+   { detail::alignment_file_input_format_exposer<t>; }
    // std::same_as<decltype(t::file_extensions), std::vector<std::string>>;
-   { v.read_alignment_record(stream,
+   {detail::alignment_file_input_format_exposer<t>{}.read_alignment_record(stream,
                              options,
                              ref_sequences,

```

Reply:
That won't work either. As I described in the comment the concept check hard fails if the using declaration fails. That is not an instantiation error as you would get with a function template because the using declaration is not used as a SFINAE expression. It doesn't matter were you put it into the concept. So, yes it is very ugly but I see no easy solution for it. A simpler interface would make it more bearable but that is another story. I could try to use some fancy SFINAE stuff inside of the exposer, but that won't make it more readable nor am I sure that it will work. The problem here is that the member function is protected and can't be used in the top-level but only within the exposer type.